### ISSUE TITLE
Healthcheck unhealthy on multi-version server

### ISSUE THREAD

**Issue by voruti** [itzg/docker-minecraft-server#796]
I have three servers with type PAPER. All of them are working fine, but the healthchecks say they are "unhealthy".

**Screenshot from Portainer:**

![portainer-screenshot](https://user-images.githubusercontent.com/45130096/110531661-22551200-811c-11eb-8b26-e628985616a1.png)

**docker-compose.yml:**
```
networks:
  default:
    external:
      name: bungeecord_connector
services:
  vanilla:
    container_name: vanilla
    environment:
      EULA: "TRUE"
      TYPE: PAPER
    image: itzg/minecraft-server:latest
    stdin_open: true
    tty: true
    volumes:
    - ./data:/data:rw
    - /etc/timezone:/etc/timezone:ro
version: '3.0'
```

**Comment by Audition-CSBlock** [comment]
@voruti sorry for getting your hopes up by commenting but could you please tell me how to put portainer in dark mode? i can't find the setting

**Comment by itzg** [comment]
I wasn't able to recreate this issue with the previous image revision, but I did notice I was behind on upgrading mc-monitor. FWIW, creating a fresh instance with

```
docker run -it --rm -e EULA=TRUE -e TYPE=PAPER -p 25565:25565 --name paper itzg/minecraft-server
```

reports healthy
```
CONTAINER ID   IMAGE                   COMMAND    CREATED              STATUS                        PORTS                                 NAMES
83207854e749   itzg/minecraft-server   "/start"   About a minute ago   Up About a minute (healthy)   0.0.0.0:25565->25565/tcp, 25575/tcp   paper
```

Do you have any plugins installed? If so, you may need to try process of elimination to see which one is corrupting the ping response.

**Comment by voruti** [comment]
> Do you have any plugins installed? If so, you may need to try process of elimination to see which one is corrupting the ping response.

Found it. It indeed is because of the plugin [ProtocolSupport](https://www.spigotmc.org/resources/protocolsupport.7201/) which I use on all of the servers. The Minecraft-Version of the servers report as: "1.8.x, 1.9.x, 1.10.x, 1.11.x, 1.12.x, 1.13.x, 1.14.x, 1.15.x, 1.16.x" (https://mcsrvstat.us/server/main.m1ne.de).

> ![portainer-screenshot](https://user-images.githubusercontent.com/45130096/110531661-22551200-811c-11eb-8b26-e628985616a1.png)

This makes sense, because the healthcheck fails with the message "Go struct field Version.version.protocol". It tries to find one single version on which the server is running, which my server is not.

**Comment by itzg** [comment]
Hmm, that's an interesting scenario for them; however, either way the server list ping response isn't compliant with the JSON structure

https://wiki.vg/Server_List_Ping#Response

**Comment by itzg** [comment]
...I wonder why they don't put that multi-version list in `version.name` rather than setting `version` to `-1`?

**Comment by Audition-CSBlock** [comment]
> > @voruti sorry for getting your hopes up by commenting but could you please tell me how to put portainer in dark mode? i can't find the setting
> 
> [Dark Reader](https://addons.mozilla.org/de/firefox/addon/darkreader/)

Danke shun!

**Comment by voruti** [comment]
I did some testing:

Basic vanilla server:
`sudo docker container run --rm -e EULA=true -p 25561:25565 itzg/minecraft-server`:
```json
{
  "description": {
    "text": "A Vanilla Minecraft Server powered by Docker"
  },
  "players": {
    "max": 20,
    "online": 0
  },
  "version": {
    "name": "1.16.5",
    "protocol": 754
  }
}
```
Basic Paper server:
`sudo docker container run --rm -e EULA=true -p 25561:25565 -e TYPE=PAPER itzg/minecraft-server`:
```json
{
  "description": {
    "text": "A Paper Minecraft Server powered by Docker"
  },
  "players": {
    "max": 20,
    "online": 0
  },
  "version": {
    "name": "Paper 1.16.5",
    "protocol": 754
  }
}
```
Difference: "Paper " is added to version.name

Paper server with plugin ProtocolSupport installed on it:
`sudo docker container run --rm -e EULA=true -p 25561:25565 -e TYPE=PAPER -v /home/user/tmpWorkDir:/data itzg/minecraft-server` (server started, Ctrl+C, moved ProtocolSupport.jar to /home/user/tmpWorkDir/plugins/ProtocolSupport.jar, server started again):
```json
{
  "version": {
    "name": "Paper 1.16.5",
    "protocol": 4
  },
  "description": {
    "text": "A Paper Minecraft Server powered by Docker"
  },
  "players": {
    "max": 20,
    "online": 0
  }
}
```
Difference: version.protocol changed from 754 to 4 & the version block moved to the beginning of the JSON structure (this is reproducible!)

So, ... the version.name value is not changed by the plugin. It's the version.protocol that is changed to 4.

**Comment by voruti** [comment]
The information from https://github.com/itzg/docker-minecraft-server/issues/796#issuecomment-796880533 was collected with https://gist.github.com/zh32/7190955 which is linked in https://wiki.vg/Server_List_Ping#Examples.
I did the same with https://github.com/Raqbit/mc-pinger (which is used by mc-monitor, which is used by docker-minecraft-server): The JSON response from my server running the ProtocolSupport plugin is
```json
{
  "description": {
    "text": "A Vanilla Minecraft Server powered by Docker"
  },
  "players": {
    "max": 8,
    "online": 0
  },
  "version": {
    "name": "Paper 1.16.5",
    "protocol": -1
  }
}
```
(Ignoring the modt and player count, which are different from the previous comment, because this JSON response is from my "production" server, e. g. further configured server ...)
... a difference in the version.protocol can be seen! There's that "-1" from the error message in the screenshot I attached in my first https://github.com/itzg/docker-minecraft-server/issues/796#issue-826576960.

**Comment by voruti** [comment]
Now for the solution :D

When I modified https://github.com/Raqbit/mc-pinger to extract the JSON response, I used the latest version from the master branch (git clone). No error is showing up!!
[iverly](https://github.com/iverly) just pushed a [commit](https://github.com/Raqbit/mc-pinger/commit/1229469434c0f37d12bf5aa0821f0b89163894f9) which fixed the problem. He changed the "Protocol" variable from uint to int32 which allows the variable to contain negative values and thus the error is fixed with this newest commit.

With this change integrated into this Docker image "itzg/minecraft-server", the healthcheck should be fixed!
